internal/tools/k8s/helpers.go

```diff
@@ -2,27 +2,28 @@ package k8s
 
 import (
 	"errors"
-	"strings"
 	"time"
 
-	apiv1 "k8s.io/api/core/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
 )
 
 // WaitForPodRunning will poll a given kubernetes pod at a regular interval for
 // it to enter the 'Running' state. If the pod fails to become ready before
 // the provided timeout then an error will be returned.
-func (kc *KubeClient) WaitForPodRunning(name, namespace string, timeout int) (apiv1.Pod, error) {
+func (kc *KubeClient) WaitForPodRunning(namespace, podName string, timeout int) (*corev1.Pod, error) {
```

Comment:
Might we accept a `context` instead of a `timeout` instead? This would allow the client to define the cancellation semantics (whether time-based, or an explicit `cancel`, etc.)